src/components/NglViewer/NglViewer.tsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useRef } from "react";
+import * as NGL from "ngl";
+import "./NglViewer.css";
+interface ActiveStructure {
+  fileName: string;
+  downloadName: string;
+}
+interface InputType {
+  activeStructure: ActiveStructure;
+}
+const NglViewer = ({ activeStructure }: InputType) => {
+  const viewportRef = useRef<HTMLDivElement>(null);
+  const dialogRef = useRef<HTMLDialogElement>(null);
+  const stage = useRef<NGL.Stage | null>(null);
+
+  useEffect(() => {
+    if (!viewportRef.current) {
+      return;
+    }
+
+    // Create NGL Stage object
+    stage.current = new NGL.Stage(viewportRef.current);
+  }, []);
+
+  useEffect(() => {
+    if (stage.current === null) {
+      return;
+    }
+    if (activeStructure.fileName === "") {
+      dialogRef.current?.close();
+    } else {
+      dialogRef.current?.showModal();
+      stage.current.removeAllComponents();
```

Comment:
This line fixes https://github.com/haddocking/haddock3/issues/670, when https://github.com/haddocking/haddock3/pull/683 on that repo is merged we can close it.

Reply:
tested with two different files, the issue should be fixed.